### ISSUE TITLE
Progress block returning wrong value

### ISSUE THREAD

**Issue by vivekfarshore** [chasseurmic/TWRDownloadManager#36]
I am downloading the video in tableview cell using this method
    [[TWRDownloadManager sharedManager] downloadFileForURL:FILE_URL progressBlock:^(CGFloat progress) {
        self.progressView.progress = progress;
    } remainingTime:^(NSUInteger seconds) {
        NSLog(@"ETA: %lu sec.", (unsigned long)seconds);
        self.mainLabel.text = [NSString stringWithFormat:@"Progress: %.0f%% - ETA: %lu sec.",      self.progress*100, (unsigned long)seconds];
    } completionBlock:^(BOOL completed) {
        NSLog(@"Download completed!");
        self.deleteButton.enabled = YES;
        self.cancelButton.enabled = NO;
        self.startButton.enabled = NO;
    } enableBackgroundMode:YES];

**Comment by Rix1212** [comment]
Yes I also faced the same issue the CGFloat in progress block  returns negative values
